src/network/networktool.cpp

```diff
@@ -0,0 +1,178 @@
+#include "logger.h"
+#include "maddy/parser.h"
+#include "networkmanager.h"
+#include "networktool.h"
+#include "notificationmanager.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QRegularExpression>
+#include <QTimer>
+#include <QUrl>
+
+PING_LOGGING_CATEGORY(NETWORKTOOL, "ping.networktool")
+
+void NetworkTool::checkInterfaceUpdate()
+{
+    static QString gitUserRepo;
+    if(gitUserRepo.isEmpty()) {
+        //https://github.com/user/repo.git
+        const static QRegularExpression regex("github.com/(.*).git");
+        QRegularExpressionMatch regexMatch = regex.match(QStringLiteral(GIT_URL));
+        // TODO: Check regex output
+        if (regexMatch.hasMatch()) {
+            gitUserRepo = regexMatch.capturedTexts()[1];
+        } else {
+            qCWarning(NETWORKTOOL) <<
+                                   "Fail to get github user and repository! It'll not be possible to check for updates";
+            return;
+        }
+    }
+
+    const QUrl url{QStringLiteral("https://api.github.com/repos/%1/releases").arg(gitUserRepo)};
+    NetworkManager::self()->requestJson(
+        url,
+        self()->checkNewVersionInGitHubPayload
+    );
+}
+
+void NetworkTool::checkNewVersionInGitHubPayload(QJsonDocument& jsonDocument)
+{
+    const static QString projectTag = QStringLiteral(GIT_TAG);
+    static QRegularExpression releaseTagRegex;
+    // Check if version start with 't' from test version
+    // For everything we can presume normal release
+    if(projectTag.contains('t') == 0) {
```

Comment:
// Check if version starts with 't' for test version
// Check if version starts with 'v' for release version